doc/HOWTO/documenting-functions-and-macros.md

```diff
@@ -0,0 +1,176 @@
+You should git add the result as `generate_doc_buildinfo` is not run on every build.
+
+With these two changes, running `make doc-nits` locally should
+now agree with you that you have documented all your new defines,
+but it might then complain:
+
+```text
+BIO_get_dgram_dest(3) is supposedly internal
+(maybe missing from other.syms) but is documented as public
+```
+
+If it is the case that your interface is meant to be public, then you need
+to edit the file `util/other.syms` to add the names of your `#define`
+functions.
+This file gets sorted alphabetically prior to each major release,
+but new additions should be placed at the end of the file.
+
+### Example
+
+For demonstration purposes, two new public symbols have been added
+by "implementing" a public function `BIO_set_dgram_foo()`
+and a public function-like macro `BIO_set_dgram_bar()`:
+
+```diff
+diff --git a/crypto/bio/bss_dgram.c b/crypto/bio/bss_dgram.c
+index 82d382cf4e..30382f0abe 100644
+--- a/crypto/bio/bss_dgram.c
++++ b/crypto/bio/bss_dgram.c
+@@ -192,6 +192,13 @@ BIO *BIO_new_dgram(int fd, int close_flag)
+     return ret;
+ }
+
++
++int BIO_set_dgram_foo(BIO* b, int foo)
++{
++    return foo;
++}
++
++
+ static int dgram_new(BIO *bi)
+ {
+     bio_dgram_data *data = OPENSSL_zalloc(sizeof(*data));
+```
+```diff
+diff --git a/include/openssl/bio.h.in b/include/openssl/bio.h.in
+index c70185db34..4ddea2f96b 100644
+--- a/include/openssl/bio.h.in
++++ b/include/openssl/bio.h.in
+@@ -485,6 +485,9 @@ struct bio_dgram_sctp_prinfo {
+ #define BIO_set_dgram_dest(b, addr)   BIO_ctrl(b, BIO_CTRL_DGRAM_SET_PEER, 0, addr)
+ #define BIO_get_dgram_dest(b, addr)   BIO_ctrl(b, BIO_CTRL_DGRAM_GET_PEER, 0, addr)
+
+ +int BIO_set_dgram_foo(BIO* b, int foo);
+ +#define BIO_set_dgram_bar(b, bar) BIO_ctrl(b, BIO_CTRL_DGRAM_SET_ADDR, 0, bar)
+ +
+  /*
+   * name is cast to lose const, but might be better to route through a
+   * function so we can do it safely
+```
+
+If you run `make doc-nits`, you might be surprised that it only
+complains about the undocumented macro, not the function:
+
+```console
+$ make doc-nits
+
+/usr/bin/perl ./util/find-doc-nits -c -n -l -e
+include/openssl/bio.h: macro BIO_set_dgram_bar(3) undocumented
+# 1 macros undocumented (count is approximate)
+make: *** [Makefile:3833: doc-nits] Error 1
+```
+
+The explanation for this is that one important step is still missing,
+it needs to be done first: you need to run
+
+```console
+$ make update
+```
+
+which triggers a scan of the public headers for new API functions.
+
+All new functions will be added to either `util/libcrypto.num`
+or `util/libssl.num`.
+Those files store the information about the symbols which need
+to be exported from the shared library resp. DLL.
+Among other stuff, they contain the ordinal numbers for
+the [module definition file](https://docs.microsoft.com/en-us/cpp/build/exporting-from-a-dll-using-def-files) of the Windows DLL,
+which is the reason for the `.num` extension.
+
```

Comment:
You could do with this link like you did with document-policy at the start of this file, i.e. something like this:

```suggestion
All new functions will be added to either `util/libcrypto.num`
or `util/libssl.num`.
Those files store the information about the symbols which need
to be exported from the shared library resp. DLL.
Among other stuff, they contain the ordinal numbers for the
[module definition file] of the Windows DLL, which is the
reason for the `.num` extension.

[module definition file]: https://docs.microsoft.com/en-us/cpp/build/exporting-from-a-dll-using-def-files 
```

That should help fixing the line length problem.